pkg/webui/components/events/widget/index.js

```diff
@@ -0,0 +1,128 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import bind from 'autobind-decorator'
+import { defineMessages } from 'react-intl'
+
+import Link from '../../link'
+import Message from '../../../lib/components/message'
+import Icon from '../../icon'
+import Status from '../../status'
+import List from '../../list'
+import PropTypes from '../../../lib/prop-types'
+import getEventComponentByName from '../../event/types'
+
+import DateTime from '../../../lib/components/date-time'
+import style from './widget.styl'
+
+const m = defineMessages({
+  latestEvents: 'Latest events',
+  seeAllActivity: 'See all activity',
+  noEvents: '{entityId} has not sent any events recently',
+  unknown: 'Unknown',
+})
+
+@bind
+class EventsWidget extends React.PureComponent {
+
+  renderEvent (event) {
+    const { component: Component, type } = getEventComponentByName(event.name)
+
+    return (
+      <List.Item>
+        <Component
+          event={event}
+          type={type}
+        />
+      </List.Item>
+    )
+  }
+
+  render () {
+    const {
+      className,
+      events,
+      toAllUrl,
+      emitterId,
+      connectionStatus,
+    } = this.props
+
+    let latestActivityTime = null
+    if (events.length) {
+      const latestEvent = events[events.length - 1]
+      latestActivityTime = <DateTime.Relative value={latestEvent.time} />
+    } else {
+      latestActivityTime = <Message content={m.unknown} />
+    }
+
+    const statusMessage = (
+      <span>
+        <Message
+          className={style.statusMessage}
+          content={m.latestEvents}
+        />
+        {latestActivityTime}
+      </span>
+
+    )
+
+    return (
+      <aside className={className}>
+        <div className={style.header}>
+          <Status
+            label={statusMessage}
+            status={connectionStatus}
+          />
+          <Link to={toAllUrl}>
+            <Message
+              className={style.seeAllMessage}
+              content={m.seeAllActivity}
+            />
+            →
+          </Link>
+        </div>
+        <List
```

Comment:
Atm, the list will keep growing when list items overflow. It should rather have a fixed height and use `overflow: hidden` (and possibly also limit the number of items rendered, accordingly).